src/test/java/lambda/part3/exercise/Exercise2.java

```diff
@@ -0,0 +1,138 @@
+package lambda.part3.exercise;
+
+import lambda.data.Employee;
+import lambda.data.JobHistoryEntry;
+import lambda.data.Person;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Function;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.contains;
+
+@SuppressWarnings({"unused", "ConstantConditions"})
+class Exercise2 {
+
+    private static class MapHelper<T> {
+
+        private final List<T> source;
+
+        private MapHelper(List<T> source) {
+            this.source = source;
+        }
+
+        public static <T> MapHelper<T> from(List<T> source) {
+            return new MapHelper<>(source);
+        }
+
+        public List<T> getMapped() {
+            return new ArrayList<>(source);
+        }
+
+        /**
+         * Создает объект для маппинга, передавая ему новый список, построенный на основе исходного.
+         * Для добавления в новый список каждый элемент преобразовывается с использованием заданной функции.
+         * ([T], (T -> R)) -> [R]
+         * @param mapping Функция преобразования элементов.
+         */
+        public <R> MapHelper<R> map(Function<T, R> mapping) {
+            ArrayList<R> newSource = new ArrayList<>();
+            source.forEach(element -> newSource.add(mapping.apply(element)));
+            return new MapHelper<>(newSource);
+        }
+
+        /**
+         * Создает объект для маппинга, передавая ему новый список, построенный на основе исходного.
+         * Для добавления в новый список каждый элемент преобразовывается в список с использованием заданной функции.
+         * ([T], (T -> [R])) -> [R]
+         * @param flatMapping Функция преобразования элементов.
+         */
+        public <R> MapHelper<R> flatMap(Function<T, List<R>> flatMapping) {
+            ArrayList<R> newSource = new ArrayList<>();
+            source.forEach(oldSourceElement -> newSource.addAll(flatMapping.apply(oldSourceElement)));
+            return new MapHelper<>(newSource);
+        }
+    }
+
+    @Test
+    void mapEmployeesToLengthOfTheirFullNamesUsingMapHelper() {
+        List<Employee> employees = getEmployees();
+
+        List<Integer> lengths = MapHelper.from(employees)
+                                  .map(Employee::getPerson)
+                                  .map(Person::getFullName)
+                                  .map(String::length)
+                                  .getMapped();
+        assertThat(lengths, contains(14, 19, 14, 15, 14, 16));
+    }
+
+    @Test
+    void mapEmployeesToCodesOfLetterTheirPositionsUsingMapHelper() {
+        List<Employee> employees = getEmployees();
+
+        List<Integer> codes = MapHelper.from(employees)
+                .flatMap(Employee::getJobHistory)
+                .map(JobHistoryEntry::getPosition)
+                .flatMap(Exercise2::calcCodes)
+                .map(c -> c)
```

Comment:
this operation seems redundant

Reply:
Yes, but it is a line in test wrote by @elefus

Reply:
According to the task flatMap(String -> Character(letter)) flatMap should return a List of Characters, not a List of Integers.

Reply:
@AnyaYu Discussed with @elefus , it is fine to do this way